enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,964 @@
+// of the failure domain, and the vCenter tag category associated with this
+// failure domain.
+type VSpherePlatformFailureDomain struct {
+  // name is the name of the vCenter tag that represents this failure domain
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=80
+  Name string `json:"name"`
+
+  // type is the name of the failure domain type, which includes
+  // Datacenter, ComputeCluster and HostGroup
+  // +kubebuilder:validation:Enum=HostGroup;Datacenter;ComputeCluster
+  // +kubebuilder:validation:Required
+  Type VSpherePlatformFailureDomainType `json:"type"`
+
+  // tagCategory is the category used for the tag
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=80
+  TagCategory string `json:"tagCategory"`
+}
+
+// VSpherePlatformTopology holds the required and optional vCenter objects - datacenter,
+// computeCluster, networks, datastore and resourcePool - to provision virtual machines.
+type VSpherePlatformTopology struct {
+  // datacenter is the vCenter datacenter in which virtual machines will be located
+  // and defined as the failure domain.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=80
+  Datacenter string `json:"datacenter"`
+
+  // computeCluster as the failure domain
+  // This is required to be a path
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=2048
+  ComputeCluster *string `json:"computeCluster,omitempty"`
+
+  // Hosts has information required for placement of machines on VSphere hosts.
+  // +kubebuilder:validation:Optional
+  Hosts *VSpherePlatformFailureDomainHosts `json:"hosts,omitempty"`
+
+  // networks is the list of networks within this failure domain
+  // +kubebuilder:validation:Optional
+  Networks []string `json:"networks,omitempty"`
+
+  // datastore is the name or inventory path of the datastore in which the
+  // virtual machine is created/located.
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=2048
+  Datastore string `json:"datastore,omitempty"`
+}
+
+type VSpherePlatformFailureDomainHosts struct {
+  // vmGroupName is the Virtual Machine Group name configured
+  // within a vCenter cluster that is associated with
+  // the corresponding Host Group.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=80
+  VMGroupName string `json:"vmGroupName"`
+
+  // hostGroupName is the Host Gorup name configured
+  // within a vCenter cluster defining a group
+  // of ESXi hosts.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=80
+  HostGroupName string `json:"hostGroupName"`
```

Comment:
is hostGroup same as host clusters in vcenter? If no - how is it different from ComputerClusters used elsewhere?

Reply:
Sounds like I need to make this more clear as well. In capv you can have a topology made up of a group of hosts within a datacenter or cluster.